Sources/XCTest/XCTestCase.swift

```diff
@@ -39,8 +44,75 @@ public class XCTestCase: XCTest {
     ///   https://bugs.swift.org/browse/SR-1129 for details.
     public var _name: String
 
-    public required override init() {
-        _name = "\(self.dynamicType).<unknown>"
+    public override var testRunClass: AnyClass {
+        return XCTestCaseRun.self
+    }
+
+    public override func performTest(run: XCTestRun) {
```

Comment:
_name = "\(self.dynamicType).\(name)"
+ self.testClosure = testClosure
+ }
+
+ /// Invoking a test performs its setUp, invocation, and tearDown. In
+ /// general this should not be called directly.
+ public func invokeTest() {
+ setUp()
+ do {
+ try testClosure(self)
+ } catch {
+ recordFailureWithDescription(
+ "threw error \"\(error)\"",
+ inFile: "<EXPR>",
+ atLine: 0,
+ expected: false)
+ }
+ tearDown()
+ }
+
+ /// Records a failure in the execution of the test and is used by all test
+ /// assertions.
+ /// - Parameter description: The description of the failure being
reported.
+ /// - Parameter filePath: The file path to the source file where the
failure
+ /// being reported was encountered.
+ /// - Parameter lineNumber: The line number in the source file at filePath
+ /// where the failure being reported was encountered.
+ /// - Parameter expected: `true` if the failure being reported was the
+ /// result of a failed assertion, `false` if it was the result of an
+ /// uncaught exception.
+ public func recordFailureWithDescription(description: String, inFile
filePath: String, atLine lineNumber: UInt, expected: Bool) {
+ testRun?.recordFailureWithDescription(
+ "\(name) : \(description)",
+ inFile: filePath,
+ atLine: lineNumber,
+ expected: expected)
+
+ // FIXME: Apple XCTest does not throw a fatal error and crash the test
+ // process, it merely prevents the remainder of a testClosure
+ // from execting after it's been determined that it has already
+ // failed. The following behavior is incorrect.
+ // FIXME: No regression tests exist for this feature. We may break it
+ // without ever realizing.
+ if !continueAfterFailure {
+ fatalError("Terminating execution due to test failure")

2016-04-03 21:14 GMT+02:00 Brian Croom <notifications@github.com>:

> In Sources/XCTest/XCTestCase.swift
> <https://github.com/apple/swift-corelibs-xctest/pull/86#discussion_r58315753>
> :
>
> > @@ -39,8 +44,75 @@ public class XCTestCase: XCTest {
> >      ///   https://bugs.swift.org/browse/SR-1129 for details.
> >      public var _name: String
> >
> > -    public required override init() {
> > -        _name = "\(self.dynamicType).<unknown>"
> > +    public override var testRunClass: AnyClass {
> > +        return XCTestCaseRun.self
> > +    }
> > +
> > +    public override func performTest(run: XCTestRun) {
>
> Swift 3 API import change: public func performTest(run: XCTestRun)
> becomes public func perform(run: XCTestRun)
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/apple/swift-corelibs-xctest/pull/86/files/651fd407f55f25eba400bf19003553f236e22083#r58315753>
>